kinesis_video_streamer/src/streamer.cpp

```diff
@@ -35,107 +35,91 @@ using namespace Aws::Kinesis;
-
-    /* Log4cplus setup for the Kinesis Producer SDK */
-    std::string log4cplus_config;
-    parameter_reader_->ReadParam(
-      GetKinesisVideoParameter(kStreamParameters.log4cplus_config), log4cplus_config);
-    if (!log4cplus_config.empty()) {
-      log4cplus::PropertyConfigurator::doConfigure(log4cplus_config);
-    } else {
-      log4cplus::BasicConfigurator configurator;
-      configurator.configure();
-    }
+  parameter_reader_ = std::make_shared<Ros1NodeParameterReader>();
+  subscription_installer_ = std::make_shared<RosStreamSubscriptionInstaller>(*this);
+
+  /* Log4cplus setup for the Kinesis Producer SDK */
+  std::string log4cplus_config;
+  parameter_reader_->ReadParam(
+    GetKinesisVideoParameter(kStreamParameters.log4cplus_config), log4cplus_config);
+  if (!log4cplus_config.empty()) {
+    log4cplus::PropertyConfigurator::doConfigure(log4cplus_config);
+  } else {
+    log4cplus::BasicConfigurator configurator;
+    configurator.configure();
   }
+}
 
-  KinesisManagerStatus Initialize()
-  {
-    ClientConfigurationProvider configuration_provider(parameter_reader_);
-    ClientConfiguration aws_sdk_config =
-      configuration_provider.GetClientConfiguration();
-    /* Set up subscription callbacks */
-    if (!subscription_installer_->SetDefaultCallbacks()) {
-      AWS_LOG_FATAL(__func__, "Failed to set up subscription callbacks.");
-      return KINESIS_MANAGER_STATUS_ERROR_BASE;
-    }
-    auto kinesis_client = std::unique_ptr<KinesisClient>(
-      Aws::New<Aws::Kinesis::KinesisClientFacade>(__func__, aws_sdk_config));
-    stream_manager_ = std::make_shared<KinesisStreamManager>(
-      parameter_reader_.get(), &stream_definition_provider_, subscription_installer_.get(),
-      std::move(kinesis_client));
-    subscription_installer_->set_stream_manager(stream_manager_.get());
-    /* Initialization of video producer */
-    KinesisManagerStatus initialize_video_producer_result =
-      stream_manager_->InitializeVideoProducer(aws_sdk_config.region.c_str());
-    if (KINESIS_MANAGER_STATUS_FAILED(initialize_video_producer_result)) {
-      AWS_LOGSTREAM_FATAL(__func__, "Failed to initialize video producer. Error code: "
-                                      << initialize_video_producer_result);
-      return initialize_video_producer_result;
-    }
-    /* Set up subscriptions and get ready to start streaming */
-    KinesisManagerStatus streamer_setup_result = stream_manager_->KinesisVideoStreamerSetup();
-    if (KINESIS_MANAGER_STATUS_SUCCEEDED(streamer_setup_result)) {
-      AWS_LOG_DEBUG(__func__, "KinesisVideoStreamerSetup completed successfully.");
-    } else {
-      AWS_LOGSTREAM_ERROR(__func__, "KinesisVideoStreamerSetup failed with error code : "
-                                      << streamer_setup_result << ". Exiting");
-      return streamer_setup_result;
-    }
-    return KINESIS_MANAGER_STATUS_SUCCESS;
+KinesisManagerStatus StreamerNode::Initialize()
```

Comment:
Consider putting the subscription_installer argument here. The subscription_installer_ gets used in this function and if setSubscriptionInstaller gets called after initialize, nothing in this class is actually affected. Or use a builder class/function and make SetSubscriptionInstaller private, this way someone does not use SetSubscriptionInstaller out of order.